### PR TITLE
Update to latest version

### PR DESCRIPTION
**Pull request by eneufeld**

### CODE REVIEW

**Comment by eneufeld** [comment]
Fix #375 
Fix #405

**Review by ndoschek — CHANGES_REQUESTED** [review]
Thanks Eugen for the update!
In general, all products still work as expected but I spotted some issues regarding the diagram styling and overall usage of icons:

1. The workflow diagram seems to have styling issues, in my case it looks like this:
image

2. If I run the client build, my `yarn.lock` is updated and the Theia dependencies are downgraded to `1.20.0` due to the resolutions in the main`package.json`. Could you please update either one of them? Thanks!

3. With Theia >= 1.18.0 the usage of `font-awesome` icons was replaced by `codicon`s. In general, to match the style of Theia we should use `codicons` as well wherever possible. E.g. for GLSP, the diagram icon (for the widget tab and the file node in the navigator) was formerly `fa fa-project-diagram` and was replaced by `codicon codicon-type-hierarchy-sub`.
Especially the `Getting started` widget needs an update of its icons. 
But in general, `codicons` do not offer the whole variety of `font-awesome` icons yet, but it would be great to find replacements from the `codicon` set.
See also the current `codicon` catalogue: https://microsoft.github.io/vscode-codicons/dist/codicon.html
image

**Comment by eneufeld** [comment]
thanks for the review, I will look at the issues

File: web/coffee-workflow-glsp-editor/src/common/workflow-language.ts [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
  ********************************************************************************/
 import { GLSPDiagramLanguage } from '@eclipse-glsp/theia-integration';
 
 export const WorkflowNotationLanguage: GLSPDiagramLanguage = {
     contributionId: 'WorkflowNotation',
     label: 'Workflow Notation diagram',
     diagramType: 'workflow-diagram-notation',
-    iconClass: 'fa fa-project-diagram',
+    iconClass: 'codicon codicon-type-hierarchy-sub',
```
Comment (ndoschek) [inline web/coffee-workflow-glsp-editor/src/common/workflow-language.ts:17]: In this case we cannot use a util function, due to the location in the `common` directory.

File: web/coffee-workflow-glsp/css/temporary_glsp.css [hunk 2]
```diff
@@ -0,0 +1,385 @@
+/********************************************************************************
```
Comment (ndoschek) [inline web/coffee-workflow-glsp/css/temporary_glsp.css:1]: Do we still need that?

File: web/coffee-workflow-glsp/src/di.config.ts [hunk 3]
```diff
@@ -9,141 +9,72 @@
  * SPDX-License-Identifier: EPL-2.0 OR MIT
  ********************************************************************************/
 import '../css/diagram.css';
+// TODO remove when GLSP is updated
+import '../css/temporary_glsp.css';
```
Comment (ndoschek) [inline web/coffee-workflow-glsp/src/di.config.ts:13]: Do we still need that?

**Review by ndoschek — CHANGES_REQUESTED** [review]
Thanks a lot for the update Eugen, looks great to me now!

However, i have some minor comments inline and two more general remarks:
- The initial layout of the diagram could be improved imho as it is quite crooked on opening the diagram editor. If using the auto layout feature, moving all nodes into the center of the canvas and deleting the bend points should restore the "old" layout rather quickly. In general we could think about deleting bend points on auto layouting - but maybe in a separate issue, wdyt?
- Please have another look at the changed files and update the headers (esp year range) there (e.g. the `*OperationHandler`s or the client files).

File: web/.prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
```
[the remark this answers is not in the document]
Comment (eneufeld) [reply web/.prettierrc.json:1]: yes I would love to do so in a second step after the update, we must discuss whether we reuse the glsp config or create an emflcoud one, fine by you?

File: web/coffee-workflow-glsp/css/temporary_glsp.css (hunk 2 above)
Comment (eneufeld) [reply web/coffee-workflow-glsp/css/temporary_glsp.css:1]: yes, this is not part of glsp 0.9 but was added after the release

File: web/coffee-workflow-glsp/src/di.config.ts (hunk 3 above)
Comment (eneufeld) [reply web/coffee-workflow-glsp/src/di.config.ts:13]: yes, this is not part of glsp 0.9 but was added after the release

File: web/.prettierrc.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (ndoschek) [reply web/.prettierrc.json:1]: Yes that sounds good to me!

File: backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java [hunk 5]
```diff
@@ -78,10 +82,10 @@
 		// Delete action
 		if (selectedElements.size() == 1) {
 			actions.add(new LabeledAction("Delete",
-					Lists.newArrayList(new DeleteOperation(editorContext.getSelectedElementIds())), "fa-minus-square"));
+					Lists.newArrayList(new DeleteOperation(editorContext.getSelectedElementIds())), "codicon-diff-removed"));
```
Comment (ndoschek) [inline backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java:85]:
```suggestion
					Lists.newArrayList(new DeleteOperation(editorContext.getSelectedElementIds())), "diff-removed"));
```

File: backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java [hunk 6]
```diff
@@ -78,10 +82,10 @@
...
 		} else if (selectedElements.size() > 1) {
 			actions.add(new LabeledAction("Delete All",
-					Lists.newArrayList(new DeleteOperation(editorContext.getSelectedElementIds())), "fa-minus-square"));
+					Lists.newArrayList(new DeleteOperation(editorContext.getSelectedElementIds())), "codicon-diff-removed"));
```
Comment (ndoschek) [inline backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java:88]:
```suggestion
					Lists.newArrayList(new DeleteOperation(editorContext.getSelectedElementIds())), "diff-removed"));
```

File: backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java [hunk 7]
```diff
@@ -99,12 +103,12 @@
 	private LabeledAction createWeightedEdgeAction(final String label, final GNode source, final GNode node) {
 		return new LabeledAction(label,
 				Lists.newArrayList(new CreateEdgeOperation(ModelTypes.WEIGHTED_EDGE, source.getId(), node.getId())),
-				"fa-plus-square");
+				"codicon-diff-added");
```
Comment (ndoschek) [inline backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java:106]:
```suggestion
				"diff-added");
```

File: backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java [hunk 8]
```diff
@@ -99,12 +103,12 @@
...
 	}
 
 	private LabeledAction createEdgeAction(final String label, final GNode source, final GNode node) {
 		return new LabeledAction(label, Lists.newArrayList(new CreateEdgeOperation(EDGE, source.getId(), node.getId())),
-				"fa-plus-square");
+				"codicon-diff-added");
```
Comment (ndoschek) [inline backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/WorkflowCommandPaletteActionProvider.java:111]:
```suggestion
				"diff-added");
```

File: web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-label-provider-contribution.ts [hunk 9]
```diff
@@ -19,44 +19,45 @@ import { CoffeeTreeEditorConstants } from './coffee-tree-editor-widget';
...
-    [CoffeeModel.Type.ManualTask, 'fa-wrench ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.Merge, 'fa-chevron-down ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.Node, 'fa-circle ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.Processor, 'fa-microchip ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.RAM, 'fa-memory ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.Task, 'fa-tasks ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.WaterTank, 'fa-tint ' + DEFAULT_COLOR],
-    [CoffeeModel.Type.WeightedFlow, 'fa-exchange-alt light-orange'],
-    [CoffeeModel.Type.Workflow, 'fa-random ' + DEFAULT_COLOR]
+    [CoffeeModel.Type.AutomaticTask, 'codicon-gear ' + DEFAULT_COLOR],
```
Comment (ndoschek) [inline web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-label-provider-contribution.ts:22]: Please also update the icon strings here. E.g:
```suggestion
    [CoffeeModel.Type.AutomaticTask, 'gear ' + DEFAULT_COLOR],
```

File: web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-label-provider-contribution.ts [hunk 10]
```diff
@@ -19,44 +19,45 @@ import { CoffeeTreeEditorConstants } from './coffee-tree-editor-widget';
...
+    [CoffeeModel.Type.BrewingUnit, 'codicon-flame ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.ControlUnit, 'codicon-server ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Decision, 'codicon-chevron-up ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Dimension, 'codicon-move ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.DipTray, 'codicon-inbox ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Display, 'codicon-tv ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Flow, 'codicon-arrow-swap ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Fork, 'codicon-source-control codicon-rotate-90 ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Join, 'codicon-source-control codicon-rotate-270 ' + DEFAULT_COLOR],
```
Comment (ndoschek) [inline web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-label-provider-contribution.ts:31]: Do we even support `Fork` and `Join` nodes? I can see that it is available in the `Coffee.ecore` and on the client side (`coffee-workflow-glsp/src/model.ts`) but we do not offer a creation tool or a `ModelType` on the server side.

File: web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-label-provider-contribution.ts [hunk 11]
```diff
@@ -19,44 +19,45 @@ import { CoffeeTreeEditorConstants } from './coffee-tree-editor-widget';
...
+    [CoffeeModel.Type.RAM, 'codicon-memory ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.Task, 'codicon-checklist ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.WaterTank, 'codicon-beaker ' + DEFAULT_COLOR],
+    [CoffeeModel.Type.WeightedFlow, 'codicon-arrow-swap light-orange'],
+    [CoffeeModel.Type.Workflow, 'codicon-combine ' + DEFAULT_COLOR]
 ]);
 
 /* Icon for unknown types */
-const UNKNOWN_ICON = 'fa-question-circle ' + DEFAULT_COLOR;
+const UNKNOWN_ICON = 'codicon-circle-slash ' + DEFAULT_COLOR;
```
Comment (ndoschek) [inline web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-label-provider-contribution.ts:45]:
```suggestion
const UNKNOWN_ICON = 'circle-slash ' + DEFAULT_COLOR;
```

**Review by ndoschek — CHANGES_REQUESTED** [review]
Thanks again Eugen, I just spotted some minor oversights, but my requests are mostly resolved, thanks for that!
One comment from before is unresolved: https://github.com/eclipse-emfcloud/coffee-editor/pull/424#discussion_r800571902
Could you have another look? Thanks a lot!

**Review by ndoschek — APPROVED** [review]
Thanks a lot Eugen, looks great to me now! :tada:

**Pull request merged**